Modules/_tkinter.c

```diff
@@ -589,10 +589,14 @@ Tkapp_New(const char *screenName, const char *className,
           int interactive, int wantobjects, int wantTk, int sync,
           const char *use)
 {
+    PyTypeObject *type;
     TkappObject *v;
     char *argv0;
 
-    v = PyObject_New(TkappObject, (PyTypeObject *) Tkapp_Type);
+    type = (PyTypeObject *)Tkapp_Type;
+    assert(type != NULL);
+    assert(type->tp_alloc != NULL);
+    v = (TkappObject *)type->tp_alloc(type, 0);
```

Comment:
I would prefer to move the variable definition aside to their first assignment:

```suggestion
    PyTypeObject *type = (PyTypeObject *)Tkapp_Type;
    assert(type != NULL);
    assert(type->tp_alloc != NULL);
    TkappObject *v = (TkappObject *)type->tp_alloc(type, 0);
```

Same below.

Reply:
I followed the existing style. Do you want me to move the `char *arg` as well?

Reply:
> Do you want me to move the char *arg as well?

As you want. You can leave *argv0* as it is to keep the diff small.

Reply:
I think I would me comfortable not changing the style of this function. It's a long function, and if we were to use gotos instead for some refactoring, we could benefit for having everything declared at the top. Do you mind I leave it as is?